app/views/users/new.html.erb

```diff
@@ -50,3 +50,55 @@
     <% end %>
   </div>
 </div>
+
+<script>
+const el/*HTMLInputElement*/ = document.getElementById("user_display_name")
+const isTaken/*Record<string, string>*/ = {}
+
+function giveFeedback(username /* string*/, isValid /*boolean | undefined */) {
+  if (el?.value !== username) {
+    return
+  }
+  if(isValid === undefined){
+    el.classList.remove("is-invalid")
+    el.classList.remove("is-valid")
+    return
+  }
+  if (isValid) {
+    el.classList.add("is-valid")
+    el.classList.remove("is-invalid")
+  } else {
+    el.classList.add("is-invalid")
+    el.classList.remove("is-valid")
+  }
```

Comment:
```js
el.classList.toggle("is-valid", isValid)
el.classList.toggle("is-invalid", !isValid)
```

or `$("#user_display_name").toggleClass("is-valid", isValid)` etc if we're still using jquery